### PR TITLE
Raise error if IPPool is created with a bad cidr block size

### PR DESCRIPTION
**Pull request by TrimBiggs**
Fixes https://github.com/projectcalico/calico-containers/issues/505

### CODE REVIEW

File: calico_containers/pycalico/datastore_datatypes.py [hunk 1]
```diff
@@ -133,9 +134,16 @@ def __init__(self, cidr, ipip=False, masquerade=False, ipam=True):
         """
         # Normalize the CIDR (e.g. 1.2.3.4/16 -> 1.2.0.0/16)
         self.cidr = IPNetwork(cidr).cidr
+        self.ipam = bool(ipam)
+        if self.ipam:
```
Comment (robbrockbank) [inline calico_containers/pycalico/datastore_datatypes.py:141]: You should add a comment to the doc string to indicate the limitation on the CIDR when used with calico IPAM module

**Comment by robbrockbank** [comment]
Minor comment and then LGTM.

File: calico_containers/pycalico/datastore_datatypes.py (hunk 1 above)
Comment (TrimBiggs) [inline calico_containers/pycalico/datastore_datatypes.py:141]: Added

**Comment by TrimBiggs** [comment]
Merging once tests pass.

File: calico_containers/pycalico/block.py [hunk 2]
```diff
@@ -543,3 +543,9 @@ class AddressNotAssignedError(BlockError):
     Tried to query an address that isn't assigned.
     """
     pass
+
+class AddressRangeNotAllowedError(BlockError):
```
Comment (robbrockbank) [inline calico_containers/pycalico/block.py]: Not sure I like the name of this.  It isn't a range. it's a CIDR.

Perhaps just CIDRToSmallError or something along those lines.

**Comment by robbrockbank** [comment]
LGTM

**Pull request merged**